wp/lib/bap_wp/src/cfg_path.ml

```diff
@@ -0,0 +1,106 @@
+(***************************************************************************)
+(*                                                                         *)
+(*  Copyright (C) 2018/2019 The Charles Stark Draper Laboratory, Inc.      *)
+(*                                                                         *)
+(*  This file is provided under the license found in the LICENSE file in   *)
+(*  the top-level directory of this project.                               *)
+(*                                                                         *)
+(*  This work is funded in part by ONR/NAWC Contract N6833518C0107.  Its   *)
+(*  content does not necessarily reflect the position or policy of the US  *)
+(*  Government and no official endorsement should be inferred.             *)
+(*                                                                         *)
+(***************************************************************************)
+
+open !Core
+open Bap.Std
+open Graphlib.Std
+
+module Constr = Constraint
+module Expr = Z3.Expr
+module Bool = Z3.Boolean
+module BV = Z3.BitVector
+module Model = Z3.Model
+module Solver = Z3.Solver
+
+module Node = Graphs.Ir.Node
+module Edge = Graphs.Ir.Edge
+module NodeSet = Graphs.Ir.Node.Set
+module EdgeSet = Graphs.Ir.Edge.Set
+
+let add_edges (s : EdgeSet.t) (es : Edge.t list) : EdgeSet.t =
+  List.fold es ~f:EdgeSet.add ~init:s
+
+let jmps_of_blk (b : Blk.t) : Jmp.t seq =
+  Term.enum jmp_t b
+
+let jmps_of_sub (s : Sub.t) : Jmp.t seq =
+  Term.enum blk_t s |> Seq.map ~f:jmps_of_blk |> Seq.concat
+
+let filter_path (p : Constr.path) (jmps : Jmp.t seq) : Constr.path =
+  Jmp.Map.filter_keys p ~f:(fun k -> Seq.mem jmps k ~equal:Jmp.equal)
+
+let pick_edges (e1 : Edge.t) (e2 : Edge.t) (path : Constr.path) : Edge.t list =
+  let find e = Jmp.Map.find path (Edge.jmp e) in
+  let branches_taken, true_edge, false_edge =
+    match find e1, find e2 with
+    | Some bs, None -> bs, e1, e2
+    | None, Some bs -> bs, e2, e1
+    | _, _ -> failwith "expected one guarded and one fallthrough edge"
+  in
+  let open Constr in
+  match branches_taken with
```

Comment:
Minor style suggestion, you can do `match (branches_taken : Constr.branches) with` instead of doing a local `open`.

Reply:
Removed local `open`.